source/dub/generators/generator.d

```diff
@@ -668,6 +668,62 @@ ProjectGenerator createProjectGenerator(string generator_type, Project project)
 }
 
 
+/**
+	Calls delegates on files and directories in the given path that match any globs.
+*/
+void findFilesMatchingGlobs(in NativePath path, in string[] globList, void delegate(string file) addFile, void delegate(string dir) addDir)
+{
+	import std.path : globMatch;
+
+	string[] globs;
+	foreach (f; globList)
+	{
+		if (f.canFind("*", "?") ||
+			(f.canFind("{") && f.balancedParens('{', '}')) ||
+			(f.canFind("[") && f.balancedParens('[', ']')))
+		{
+			globs ~= f;
+		}
+		else
+		{
+			if (f.isDir)
+				addDir(f);
+			else
+				addFile(f);
+		}
+	}
+	if (globs.length) // Search all files for glob matches
+		foreach (f; dirEntries(path.toNativeString(), SpanMode.breadth))
+			foreach (glob; globs)
+				if (f.name().globMatch(glob))
+				{
+					if (f.isDir)
+						addDir(f);
+					else
+						addFile(f);
+					break;
+				}
+}
+
+
+/**
+	Calls delegates on files in the given path that match any globs.
+
+	If a directory matches a glob, the delegate is called on all existing files inside it recursively
+	in depth-first pre-order.
```

Comment:
https://run.dlang.io/is/cvvhB9 Comment says depth-first, all span modes are by breadth?

Reply:
The comment follows the documentation of [breadth](https://dlang.org/phobos/std_file.html#breadth.) Seems correct to me looking at the output of your test run?

Reply:
OK - I wasn't familiar with the SpanMode semantics